Bug/WP-24: Disabling Google Drive Integration

jalowe13 opened the pull request:
## Overview
Updated Task: Disabling Google Drive Integration as it is blocked by [WP-342](https://tacc-main.atlassian.net/browse/WP-342)
Original Task: Selecting a file in Google Drive to copy, allows you to try to copy to google drive which fails.  We need to disable copying to google drive.

## Related

* [WP-24](https://tacc-main.atlassian.net/browse/WP-24)

## Changes
- Disabled Google Drive setup integration on the front end temporarily 
- Disabled Google Drive in the Data Files dropdown menu
- Added data length check for if there are no other 3rd party app integration settings to display 
"No integrations available."
- Adjusted tests to conform the above two being disabled
- Adjusted fixture and DataListingCells call for Google Drive to be disabled


## Testing
For general testing
1. Navigate to Data Files and make sure Google Drive is not a listed system.
2. Try to copy a file and make sure that Google Drive is not a shared system in the dropdown.

For Unit Tests
1. Navigate to `Core-Portal/client/src/components/ManageAccount`
3. Run `npx vitest run` to run local component tests relating to Google Drive
4. Run `npm run test` to run all frontend tests

## UI
Before
![image](https://github.com/user-attachments/assets/f1b6c639-4408-4a7e-8b71-396f5a0bbe0e)
![image](https://github.com/user-attachments/assets/088d7c21-c689-4aa0-aec1-bf16fe3dae2b)

After
![image](https://github.com/user-attachments/assets/89708367-7b66-4e8e-ad79-edbc9b6b4807)
![image](https://github.com/user-attachments/assets/62ee8105-bec9-4890-891e-cf65a5c372b2)

fnets commented on client/src/components/DataFiles/DataFilesListing/DataFilesListingCells.jsx:
```diff
@@ -55,10 +55,11 @@ export const FileNavCell = React.memo(
-        window.open(href, '_blank');
-        return;
-      }
+      // Temporary Integrations Filtering and conditional rendering of Google Drive: WP-24
+      // if (api === 'googledrive') {
+      //   window.open(href, '_blank');
+      //   return;
+      // }
```
```suggestion
```
You can go ahead and delete this code instead of commenting it out. Git history will save the changes and let us know what they were.

fnets commented on client/src/components/DataFiles/fixtures/DataFiles.systems.fixture.js:
```diff
@@ -56,14 +56,15 @@ const systemsFixture = {
-        icon: null,
-        integration: 'portal.apps.googledrive_integration',
-      },
+      // Temporary Integrations Filtering and conditional rendering of Google Drive: WP-24
+      // {
+      //   name: 'Google Drive',
+      //   system: 'googledrive',
+      //   scheme: 'private',
+      //   api: 'googledrive',
+      //   icon: null,
+      //   integration: 'portal.apps.googledrive_integration',
+      // },
```
```suggestion
```

fnets reviewed (changes requested):
A couple of changes, but so far looks good!

jalowe13 commented:
> A couple of changes, but so far looks good!

Comments removed! I'll keep that in mind for next time

fnets commented:
So far, this is looking great and works locally! Great job! Two quick pieces of feedback:

1. On these PRs, the steps for testing are meant to make sure that when we run the changes locally, we know what to do to make sure that the changes are running correctly and exist. In this case, it'd be more like "Navigate to the Manage Account page (at blah.com/manageaccount url) and make sure that Google Drive is no longer available for integration (it should appear like the UI screenschots below)". PR reviewers should be running your unit tests anyway, but it's definitely a helpful reminder.

2. There's a way to access Google Drive that you might not be aware of. The [Copy Modal](https://github.com/TACC/Core-Portal/blob/7deffb0d4ca58f6991bbe84760b877af5b309cf2/client/src/components/DataFiles/DataFilesModals/DataFilesCopyModal.jsx#L19) allows users to copy data INTO Google Drive. And, if it is disabled, it will prompt them to connect to Google Drive. I think your filter solution paradigm will work great here.
![Screenshot 2024-10-28 at 12 30 02 PM](https://github.com/user-attachments/assets/2419f6c0-c382-494a-8ff9-63b8d7067819)

fnets reviewed (dismissed):
All this is working great and is super readable! I found one more spot where Google Drive is available and explained in my last comment.

jalowe13 commented:
> All this is working great and is super readable! I found one more spot where Google Drive is available and explained in my last comment.

Thank you for the review Frank! I'll go ahead and look into the Copy Modal and use the same method to fix that as soon as possible! I'll also adjust my testing description accordingly.

fnets reviewed (approved):
LGTM and works locally! Great job!

edmondsgarrett reviewed (approved):
LGTM, nice work!

rstijerina reviewed (dismissed):
Good work on implementation, but @fnets @edmondsgarrett I don't think this is quite necessary. We should be able to just remove the `googledrive` system entry from each portal's `_PORTAL_DATAFILES_STORAGE_SYSTEMS` field in their `settings_custom.py` files in Core-Portal-Deployments

fnets commented:
> Good work on implementation, but @fnets @edmondsgarrett I don't think this is quite necessary. We should be able to just remove the `googledrive` system entry from each portal's `_PORTAL_DATAFILES_STORAGE_SYSTEMS` field in their `settings_custom.py` files in Core-Portal-Deployments

Yep, you're totally right. I had assumed there was a reason that we hadn't done that.

jalowe13 commented:
Reverted changes! Simple that its configured through the settings_custom.py. 3rd party test that I added for code coverage is still in here. Let me know if that needs to be removed.

rstijerina commented on server/portal/settings/settings_custom.example.py:
```diff
@@ -112,14 +112,6 @@
         'readOnly': False,
         'hideSearchBar': False
     },
-    {
-        'name': 'Google Drive',
-        'system': 'googledrive',
-        'scheme': 'private',
-        'api': 'googledrive',
-        'icon': None,
-        'integration': 'portal.apps.googledrive_integration'
-    }
```
We can leave this in the example, as it's still the form we'd use in the future

jalowe13 replied:
Readded to custom example

rstijerina commented on client/src/components/ManageAccount/tests/ManageAccountTables.test.jsx:
```diff
@@ -154,7 +154,33 @@ describe('Third Party Apps', () => {
     );
     expect(getByText(/3rd Party Apps/)).toBeDefined();
     expect(getByText('Google Drive')).toBeDefined();
-    expect(getByText('Disconnect')).toBeDefined();
```
Is this a desired change?

jalowe13 replied:
The third party test is a desired change on my part. Just in case if there is other systems added in the future other than the Google Drive integration in the account setup section. It also covers for a code coverage case where google drive is disabled. As for queryByText this has been reverted in the most recent commit.

rstijerina replied:
Good additions 👍 . Specifically though, is `expect(getByText('Disconnect')).toBeDefined();` intended to be removed?
